lib/cached-request.js

```diff
@@ -138,6 +139,7 @@ CachedRequest.prototype.cachedRequest = function () {
   key = this.hashKey(key);
 
   //Open headers file
+  mkdirp(this.cacheDirectory);
```

Comment:
Wouldn't it make more sense for this live where `fs.writeFile` is called instead?